cloud/amazon/route53_health_check.py

```diff
@@ -0,0 +1,351 @@
+    response = conn.make_request('POST', uri, {'Content-Type': 'text/xml'}, xml_body)
+    body = response.read()
+    boto.log.debug(body)
+    if response.status not in (200, 204):
+        raise exception.DNSServerError(response.status,
+                                       response.reason,
+                                       body)
+    e = boto.jsonresponse.Element()
+    h = boto.jsonresponse.XmlHandler(e, None)
+    h.parse(body)
+    return e
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            state               = dict(required=False, choices=['present', 'absent'], default='present'),
+            ip_address          = dict(required=False),
+            port                = dict(required=False, type='int'),
+            type                = dict(required=True, choices=['HTTP', 'HTTPS', 'HTTP_STR_MATCH', 'HTTPS_STR_MATCH', 'TCP']),
+            resource_path       = dict(required=False),
+            fqdn                = dict(required=False),
+            string_match        = dict(required=False),
+            request_interval    = dict(required=False, type='int', choices=[10, 30], default=30),
+            failure_threshold   = dict(required=False, type='int', choices=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10], default=3),
+        )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')
+
+    state_in              = module.params.get('state')
+    ip_addr_in            = module.params.get('ip_address')
+    port_in               = module.params.get('port')
+    type_in               = module.params.get('type')
+    resource_path_in      = module.params.get('resource_path')
+    fqdn_in               = module.params.get('fqdn')
+    string_match_in       = module.params.get('string_match')
+    request_interval_in   = module.params.get('request_interval')
+    failure_threshold_in  = module.params.get('failure_threshold')
+
+    # Default port
+    if port_in is None:
+      if type_in in ['HTTP', 'HTTP_STR_MATCH']:
+        port_in = 80
+      elif type_in in ['HTTPS', 'HTTPS_STR_MATCH']:
+        port_in = 443
+      else:
+        module.fail_json(msg="parameter 'port' is required for 'type' TCP")
+
+    # string_match in relation with type
+    if type_in in ['HTTP_STR_MATCH', 'HTTPS_STR_MATCH']:
+        if string_match_in is None:
+            module.fail_json(msg="parameter 'string_match' is required for the HTTP(S)_STR_MATCH types")
+        elif len(string_match_in) > 255:
+            module.fail_json(msg="parameter 'string_match' is limited to 255 characters max")
+    elif string_match_in:
+        module.fail_json(msg="parameter 'string_match' argument is only for the HTTP(S)_STR_MATCH types")
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
+    # connect to the route53 endpoint
+    try:
+        conn = Route53Connection(**aws_connect_kwargs)
```

Comment:
normally I'd object to this but as boto.route53 doesn't have a `connect_to_region` method this is the best approach.